pkg/daemon/update.go

```diff
@@ -817,16 +825,47 @@ func origParentDir() string {
 	return filepath.Join("/etc", "machine-config-daemon", "orig")
 }
 
+func noOrigParentDir() string {
+	return filepath.Join("/etc", "machine-config-daemon", "noorig")
+}
+
 func origFileName(fpath string) string {
 	return filepath.Join(origParentDir(), fpath+".mcdorig")
 }
 
+func noOrigFileName(fpath string) string {
+	return filepath.Join(noOrigParentDir(), fpath+".mcdnoorig")
+}
+
+func createNoOrigFile(fpath string) error {
+	if err := os.MkdirAll(filepath.Dir(noOrigFileName(fpath)), 0755); err != nil {
+		return errors.Wrapf(err, "creating no orig parent dir: %v", err)
+	}
+	return writeFileAtomicallyWithDefaults(noOrigFileName(fpath), nil)
+}
+
 func createOrigFile(fpath string) error {
-	if _, err := os.Stat(fpath); err != nil {
-		// the file isn't there, no need to back it up
-		// we could check ENOENT only maybe?
+	if _, err := os.Stat(noOrigFileName(fpath)); err == nil {
+		// we already created the no orig file for this default file
 		return nil
 	}
+	if _, err := os.Stat(fpath); err != nil {
+		// create a noorig file that tells the MCD that the file wasn't present on disk in RHCOS
+		// so it can just remove it when deleting stale data, as opposed as restoring a file
+		// that was shipped _with_ RHCOS (e.g. a default chrony config).
+		return createNoOrigFile(fpath)
+	}
+
+	if _, err := exec.Command("rpm", "-qf", fpath).CombinedOutput(); err != nil {
```

Comment:
for files under `/etc` which are probably the ones being modified more often by users - we should check `/usr/etc` (see @jlebon comment here https://github.com/openshift/machine-config-operator/pull/1586#discussion_r398572693)

Maybe we could still use a combination of `/usr/etc/` and `rpm -qf` to do a best effort check here